packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -137,8 +137,20 @@ export const useArbTokenBridge = (
   // this prevents previous account/chains' transactions to show up in the current account
   // also makes sure the state of app doesn't get incrementally bloated with all accounts' txns loaded up till date
   useEffect(() => {
-    setPendingWithdrawalMap({})
-  }, [l1.provider, l2.provider, walletAddress])
+    if (!connector) {
```

Comment:
This change fix the issue when you would switch from wallet A to wallet B and back to wallet A. With wallet A having deposits/withdrawals. When going back to wallet A: 
- change address would trigger fetch of withdrawals (returned from SWR cache)
- pending withdrawals in `useArbTokenBridge` is reset to `{}`
- changes of SWR data would add withdrawals to the store
- because address changed, l1/l2 providers changes, and reset AGAIN pending withdrawals in `useArbTokenBridge`.

If we want to avoid being "vendor-locked" with wagmi, we might consider caching provider in `useNetworksAndSigners`

Reply:
Caching provider is definitely a good idea that we could do in a follow-up PR. I wonder if the refetch is also related to the fixes I made in this PR: https://github.com/OffchainLabs/arbitrum-token-bridge/pull/1138

Maybe we should merge that one first and see how it affects this one?